core/src/banking_stage/scheduler_messages.rs

```diff
@@ -0,0 +1,63 @@
+use {
+    crate::immutable_deserialized_packet::ImmutableDeserializedPacket,
+    solana_sdk::transaction::SanitizedTransaction, std::sync::Arc,
+};
+
+/// A unique identifier for a transaction.
+/// The lower 64 bits are based on the order the transaction entered banking stage.
+/// The upper 64 bits are the priority of the transaction.
+#[derive(Clone, Copy, Debug, Default, Hash, PartialEq, Eq, PartialOrd, Ord)]
+pub struct TransactionId(u128);
+
+impl TransactionId {
+    pub fn new(priority: u64, index: u64) -> Self {
+        Self(((priority as u128) << 64) | (index as u128))
+    }
+}
+
+/// Message: [Scheduler -> Worker]
+/// Transactions to be consumed (executed, recorded, committed)
+pub struct ConsumeWork {
+    pub ids: Vec<TransactionId>,
+    pub transactions: Vec<SanitizedTransaction>,
+}
+
+/// Message: [Worker -> Scheduler]
+/// Transactions to be forwarded to the next leader(s)
+pub struct ForwardWork {
+    pub ids: Vec<TransactionId>,
+    pub packets: Vec<Arc<ImmutableDeserializedPacket>>,
+}
+
+/// Message: [Worker -> Scheduler]
+/// Processed transactions.
+pub struct FinishedConsumeWork {
+    pub work: ConsumeWork,
+    pub retryable_indexes: Vec<usize>,
+}
+
+/// Message: [Worker -> Scheduler]
+/// Forwarded transactions.
+pub struct FinishedForwardWork {
+    pub work: ForwardWork,
+    pub successful: bool,
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_transaction_id_ordering() {
```

Comment:
quick and easy test to make sure ordering works for both the priority (more important) and index

Reply:
removed this: fe5922dc8523a6ec8941b308cc310e28138522ae